Update API docs for 2.0.9

cloneofghosts opened the pull request:
Updated the API docs to include changes from V2.0.9

* Added a note to summary and icon that they are calculated between 4am and 4am local time
* Added a note that `thunderstorm` and `hail` may be icons defined in the future
* Added a note about the `none` icon which you may want to change the wording of
* Updated OpenAPI spec to show V2.0.9

- Fixes #236
- Fixes #225
- Fixes https://github.com/Pirate-Weather/pirate-weather-ha/issues/243

cloneofghosts commented:
@alexander0042 Do we want to add documentation for #225 or is that a bug with the API?

alexander0042 commented:
Great changes! Also, and unrelated, but I hadn't realized how many updates you'd put into the HA integration, so thank you thank you thank you so much for that. 

I'll add a line to probability explaining how it can be >0 with no intensity, and merge this to main!

cloneofghosts commented:
A lot of the updates in the HA repo (same goes for this repo as well) is merging in updated dependencies. The majority of the other changes were things that people had created issues for.

Since I figured you were focusing on the V2 changes I figured out how to make updates to the integration so you could keep your focus there.

cloneofghosts commented:
Updated the docs to include information about `precipProbability` and added the multiple integration question to HA docs. It may be a good idea to have a FAQ page for the main API and move some of the stuff we added to the parameters there. It would be done in a separate PR since it'd be better to update the docs with this info first before adding a FAQ page. @alexander0042
